fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccountingProcessorHelper.java

```diff
@@ -820,11 +820,12 @@ private void createCreditJournalEntryForSavings(final Office office, final Strin
         SavingsAccountTransaction savingsAccountTransaction = null;
         ClientTransaction clientTransaction = null;
         final Long shareTransactionId = null;
-        final PaymentDetail paymentDetail = null;
+        PaymentDetail paymentDetail = null;
         String modifiedTransactionId = transactionId;
         if (StringUtils.isNumeric(transactionId)) {
             long id = Long.parseLong(transactionId);
             savingsAccountTransaction = this.savingsAccountTransactionRepository.findOne(id);
+            paymentDetail = this.savingsAccountTransactionRepository.findOne(id).getPaymentDetail();
```

Comment:
unnecessary repository call.  You can make use of savingsAccountTransaction object to retrieve payment details